temporal-sdk/src/main/java/io/temporal/internal/worker/Poller.java

```diff
@@ -168,15 +168,18 @@ public void awaitTermination(long timeout, TimeUnit unit) {
 
   @Override
   public void suspendPolling() {
-    log.info("suspendPolling: {}", this);
-    suspendLatch.set(new CountDownLatch(1));
```

Comment:
Absense of CAS here may replace and lose the latch which is already used by Pollers to block on. This way they will never be unblocked if suspentPolling was called twice.